packages/core/src/domain/conversion/bundle-modifications/__tests__/post-process-bundle.test.ts

```diff
@@ -0,0 +1,225 @@
+import { faker } from "@faker-js/faker";
+import { Condition, Patient } from "@medplum/fhirtypes";
+import { makeBundle } from "../../../../external/fhir/__tests__/bundle";
+import { buildDocIdFhirExtension } from "../../../../external/fhir/shared/extensions/doc-id-extension";
+import { snomedCodeMd } from "../../../../fhir-deduplication/__tests__/examples/condition-examples";
+import { makeCondition } from "../../../../fhir-to-cda/cda-templates/components/__tests__/make-condition";
+import * as bundleMods from "../modifications";
+import { postProcessBundle } from "../post-process";
+
+let replaceIdsMock: jest.SpyInstance;
+let addExtensionToConversionMock: jest.SpyInstance;
+let addMissingRequestsMock: jest.SpyInstance;
+let removePatientFromConversionMock: jest.SpyInstance;
+
+beforeAll(() => {
+  replaceIdsMock = jest.spyOn(bundleMods, "replaceIdsForResourcesWithDocExtension");
+  addExtensionToConversionMock = jest.spyOn(bundleMods, "addExtensionToConversion");
+  addMissingRequestsMock = jest.spyOn(bundleMods, "addMissingRequests");
+  removePatientFromConversionMock = jest.spyOn(bundleMods, "removePatientFromConversion");
+});
+afterEach(() => {
+  jest.clearAllMocks();
+});
+afterAll(() => {
+  jest.restoreAllMocks();
+});
+
+function makeFilename(patientId: string = faker.string.uuid()): string {
+  return `${faker.string.uuid()}_${patientId}_${faker.string.uuid()}.xml`;
+}
+
+function initTest() {
+  const conditionId = faker.string.uuid();
+  const patientId = faker.string.uuid();
+  const condition = makeCondition(
+    {
+      id: conditionId,
+      code: { coding: [snomedCodeMd] },
+    },
+    patientId
+  );
+  const fileName = makeFilename(patientId);
+  const documentExtension = buildDocIdFhirExtension(fileName);
+  return { patientId, condition, fileName, documentExtension };
+}
+
+describe("Checking postProcessBundle and its constituent functions", () => {
+  describe("replaceIDs", () => {
+    it("does not update the IDs if the extension is missing", () => {
+      const { patientId, condition } = initTest();
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId);
+      const updCondition = updatedBundle.entry?.[0]?.resource;
+      expect(updCondition).toBeDefined();
+      expect(updCondition?.id).toBe(condition.id);
+      expect(updCondition).toMatchObject(condition);
+    });
+
+    it("updates the ID and adds source metadata when extension is present", () => {
+      const { patientId, condition, fileName, documentExtension } = initTest();
+      condition.extension = [documentExtension];
+      const bundle = makeBundle({ entries: [condition], type: "collection" });
+
+      const updatedBundle = bundleMods.replaceIdsForResourcesWithDocExtension(bundle, patientId);
+      const updCondition = updatedBundle.entry?.[0]?.resource;
```

Comment:
ditto: check entry is present, check it has more than one item, check the first item has a resource... then, get the first entry. In case the test below fails, those steps help to know why it failed, where it broke.